daemons/controld/controld_cib.c

```diff
@@ -22,6 +22,90 @@
 // Call ID of the most recent in-progress CIB resource update (or 0 if none)
 static int latest_rsc_update = 0;
 
+// Call IDs of requested CIB replacements that won't trigger a new election
+// (used as a set of gint values)
+static GHashTable *cib_replacements = NULL;
+
+/*!
+ * \internal
+ * \brief Store the call ID of a CIB replacement that the controller requested
+ *
+ * The \p do_cib_replaced() callback function will avoid triggering a new
+ * election when we're notified of one of these expected replacements.
+ *
+ * \param[in] call_id  CIB call ID
+ *
+ * \note This function should be called after making any asynchronous CIB
+ *       request that may replace part of the nodes or status section. This may
+ *       include CIB sync calls.
```

Comment:
If the controller ever makes a synchronous replacement call in the future, then I guess we could allow a 0 call-id into the hash table to signal that we expect a replacement.